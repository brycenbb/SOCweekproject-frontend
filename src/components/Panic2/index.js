import React from "react";
import TextField from "@mui/material/TextField";
import Button from "@mui/material/Button";
import pic from '../../Assets/dont-panic-stars.png'


function Panic2({ state }) {

  function handleSubmit(e) {
    e.preventDefault();
    let topic = e.target.problemTopic.value;
    let problem = e.target.problemProblem.value;
    let tried = e.target.problemTried.value;
    console.log(topic, problem, tried);
    state({ topic: topic, problem: problem, tried: tried })
  }

  return (
    <main className="main-panic2" style={{ backgroundImage: `url(${pic})` }}>
      <form
        className="problemForm"
        onSubmit={(e) => {
          handleSubmit(e);
        }}
      >
        <div>
          <TextField
            sx={{ backgroundColor: "white", minWidth: "20rem" }}
            id="problemTopic"
            type="text"
            placeholder="What topic are you working on?"
          ></TextField>
        </div>
        <div>
          <TextField
            sx={{ backgroundColor: "white", minWidth: "20rem" }}
            id="problemProblem"
            type="text"
            placeholder="What is the problem?"
          ></TextField>
        </div>
        <div>
          <TextField
            sx={{ backgroundColor: "white", minWidth: "20rem" }}
            id="problemTried"
            type="text"
            placeholder="What have you tried so far?"
          ></TextField>
        </div>
        <Button type="submit" variant="contained">Placeholder</Button>
      </form>

      <ol className="panic2-links">
        <li>
          <a rel="noreferrer" href={`http://www.google.com/`} target="_blank">
            Google
          </a>
        </li>
        <li>
          <a
            rel="noreferrer"
            href={`http://www.freecodecamp.org/`}
            target="_blank"
          >
            freeCodeCamp
          </a>
        </li>
        <li>
          <a
            rel="noreferrer"
            href={`https://www.w3schools.com/`}
            target="_blank"
          >
            w3Schools
          </a>
        </li>
        <li>
          <a
            rel="noreferrer"
            href={`https://stackoverflow.com/`}
            target="_blank"
          >
            Stack Overflow
          </a>
        </li>
        <li>
          <a
            rel="noreferrer"
            href={`https://developer.mozilla.org/en-US/`}
            target="_blank"
          >
            MDN docs
          </a>
        </li>
        <li>
          <a
            rel="noreferrer"
            href={`https://beta.reactjs.org/`}
            target="_blank"
          >
            Beta React
          </a>
        </li>
      </ol>
    </main>
  );
}

export default Panic2;
