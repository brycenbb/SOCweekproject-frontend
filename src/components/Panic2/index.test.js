import Panic2 from "./index.js";
import request from "supertest";
import { test, expect } from "@jest/globals";

test('Renders correctly', () => {
    const response = request(Panic2)
        .get('/localhost:3000/panic2')
        .expect(200)
        .toJSON();
    console.log(response)
    expect(response).toMatchSnapshot();
});