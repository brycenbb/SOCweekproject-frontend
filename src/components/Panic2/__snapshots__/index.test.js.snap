// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Object {
  "data": undefined,
  "headers": Object {},
  "method": "GET",
  "url": "http://127.0.0.1:36967/localhost:3000/panic2",
}
`;
